app/controllers/api/v1/twilio_controller.rb

```diff
@@ -39,10 +39,20 @@ def establish_ride_zone
     def establish_user_conversation
       to_phone = PhonyRails.normalize_number(params['To'])
       from_phone = PhonyRails.normalize_number(params['From'])
-      @user = User.find_by_phone_number_normalized(from_phone)
-      if @user.nil?
-        @user = User.create!(name: '', user_type: :voter, ride_zone: @ride_zone, password: '12345678', phone_number: from_phone, phone_number_normalized: from_phone, email: "#{from_phone}@example.com")
-        # todo: user role as passenger?
+      sms_name = User.sms_name(from_phone)
+      attrs = {
+          name: sms_name,
+          user_type: :voter,
+          password: '12345678',
+          phone_number: from_phone,
+          phone_number_normalized: from_phone,
+          email: "#{Time.now.to_f}@example.com"
+      }
+      # create the user and rescue duplicate record error to find existing one
+      @user = begin
+        User.create!(attrs)
+      rescue ActiveRecord::RecordNotUnique
```

Comment:
I am pretty sure _both_ of those approaches can result in duplicate rows (as the documentation on them states) due to race conditions between the SELECT that checks and the INSERT after. Both articles explicitly say to prevent this you need to create a unique index and handle the ActiveRecord::RecordNotUnique exception. Doing find_or_create_by with the two columns would require both a rescue of RecordNotUnique (as in the PR) and a separate update_attributes with the remaining attributes (more code). Reading all the docs, this code seems the most straightforward, to me.